### DISCUSSION TITLE
Discussion about Scrolling horizontally/vertically or both? + pages? + zoom?

### DISCUSSION THREAD

**Discussion by francoisschwarzentruber** [tableaunoir/tableaunoir#92] [General]
Dear users,

**horizontal scrolling**
For the moment, the scrolling in tableaunoir is horizontally. This is the typically use for teaching. That scrolling mode is also ok for pdf slides.

**vertical scrolling**
Some users may want to take notes like on a paper. That scrolling mode suits a pdf document in A4 portrait.

**both vertical/horizontal scrolling**
I personnaly think that it is too much. People may get lost. But could be nice for a draft.

**pages**
Having several pages is nice for a pdf with several pages. But the problem is that it adds a new feature. Now people just ask "where are you in the single board?". Adding pages would lead to "on which page are you? + where are you on that page?". Have pages only is not suitable because of teaching (we need the horizontal scrolling).

**zoom**
Generally, softwares offer the possibility to zoom in and zoom out.

I think being able to switch between horizontal or vertical scrolling would be nice. I think that the notion of pages and zoom will just add some noise.
What do you think?

**Comment by tekhnee** [comment]
Dear @francoisschwarzentruber 

Thank you for soliciting views on these good questions.

> **both vertical/horizontal scrolling**
> I personnaly think that it is too much. People may get lost. But could be nice for a draft.

I believe that demand for this feature will be grow and accumulate with time, for good reason, and it might be better to solve the fundamental design challenge from the start.

A few conceivable ways of preventing disorientation with 2-D scrolling:

- A "minimap" on the right side of the view (as in some [text editors](https://atom-minimap.github.io/minimap/)).
- Cues to indicate the scrolling direction in which writing activity is currently taking place (e.g. an up arrow or a slightly brighter upper border if writing is taking place above).
- Synchronizing the scroll position on all clients via the websocket stream (so that all users are at the same vantage point).
 
> **pages**
> Having several pages is nice for a pdf with several pages. But the problem is that it adds a new feature. Now people just ask "where are you in the single board?". Adding pages would lead to "on which page are you? + where are you on that page?". Have pages only is not suitable because of teaching (we need the horizontal scrolling).

What if the page was synced across all clients, so that all users viewed the same page at any given moment?
 
> **zoom**
> Generally, softwares offer the possibility to zoom in and zoom out.

I'm generally in favor of zoom but understand why others might disagree.
